### PR TITLE
Feat/tree view real data

### PR DESCRIPTION
**Pull request by ciaranschutte**
https://github.com/icgc-argo/argo-docs/pull/62
Tree View with latest `develop` branch changes
- real data hookup

### CODE REVIEW

**Comment by ciaranschutte** [comment]
@hlminh2000 alternatively I can checkout your current tree view PR and rebase with develop

File: website/src/components/TreeView/data.js [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+    { name: 'gender', required: true },
+    { name: 'submitter_specimen_id', required: true },
+    { name: 'specimen_tissue_source', required: true },
+    { name: 'tumour_normal_designation', required: true },
+    { name: 'submitter_sample_id', required: true },
+    { name: 'sample_type', required: true },
+  ],
+  children: [
+    {
+      name: 'specimen',
```
Comment (hlminh2000) [inline website/src/components/TreeView/data.js]: is this file still used?

File: website/src/pages/dictionary/index.js [hunk 2]
```diff
@@ -327,20 +367,34 @@ function DataDictionary() {
...
+              <Display visible={selectedTab === TAB_STATE.DETAILS}>
+                <RenderDictionary
+                  schemas={filteredSchemas}
+                  menuContents={menuContents}
+                  isLatestSchema={isLatestSchema}
+                />
+              </Display>
+
+              <Display visible={selectedTab === TAB_STATE.OVERVIEW}>
+                <TreeView dictionary={dictionary} searchValue={searchValue} data={treeData} />
```
Comment (hlminh2000) [inline website/src/pages/dictionary/index.js]: `TreeView` doesn't take `dictionary` anymore eh?

File: website/src/pages/dictionary/index.js [hunk 3]
```diff
@@ -64,6 +63,7 @@ export const ModalPortal = ({ children }) => {
 };
 
 const data = require('./data.json');
+const treeData = require('./tree.json');
```
Comment (hlminh2000) [inline website/src/pages/dictionary/index.js]: just a question, maybe not scope of this PR, but is this meant to change with selected schema?

File: website/src/components/TreeView/data.js (hunk 1 above)
Comment (ciaranschutte) [reply website/src/components/TreeView/data.js]: Not actively. I'd like to leave it for now just to have a sample of what the tree is expecting.
I'll rename to dummy_data or something to make it less confusing

File: website/src/pages/dictionary/index.js (hunk 3 above)
Comment (ciaranschutte) [reply website/src/pages/dictionary/index.js]: @hlminh2000 I will address this in this PR. nice catch thanks! :)

Comment (ciaranschutte) [reply website/src/pages/dictionary/index.js]: Adding tree data to state to mirror how we load current dictionary data.
We can look at moving this data generation to runtime if need be

File: website/src/pages/dictionary/index.js [hunk 4]
```diff
@@ -64,10 +63,16 @@ export const ModalPortal = ({ children }) => {
...
 
 async function fetchDictionary(version) {
-  const response = await axios.get(`/data/schemas/${version}.json`);
-  return response.data;
+  try {
+    const dict = await axios.get(`/data/schemas/${version}.json`);
+    const tree = await axios.get(`/data/schemas/${version}_tree.json`);
+    return { dict: dict.data, tree: tree.data };
+  } catch (e) {
+    throw e;
```
Comment (hlminh2000) [inline website/src/pages/dictionary/index.js]: if just gonna throw, there's no need to catch :P

Comment (ciaranschutte) [reply website/src/pages/dictionary/index.js]: @hlminh2000 need an empty `finally` if not doing this. this is more explicit. I could have sworn you could just do a `try` though

**Pull request merged**